### ISSUE TITLE
Troubleshooting issues with Model Training and Unexpected Results in Phase 1

### ISSUE THREAD

**Issue by PremSuresh** [SangHunHan92/2K2K#11]
**Description:**
Hi, Thank you for your valuable work on this model. I have encountered an issue while training the provided model with the 2k2k dataset during Phase 1. The expected results are not being achieved, and I am seeking assistance in troubleshooting the problem.
**Examples of normal images after training phase 1 for 30 epochs:** (on 13812 training samples generated by render.py - excluded ~200 samples for inv_affine failure issue)
![000116_front](https://github.com/SangHunHan92/2K2K/assets/37246172/a24d9409-8224-461c-9240-4432c628f487)
![000116_back](https://github.com/SangHunHan92/2K2K/assets/37246172/d34039ad-d0c3-45be-833d-e9325ef84adb)
![son_front](https://github.com/SangHunHan92/2K2K/assets/37246172/71003cc0-eb1e-477f-bed4-451a04d71750)
![son_back](https://github.com/SangHunHan92/2K2K/assets/37246172/c27ebbe1-c2a1-4f64-9503-0a9aa3490fce)

**Details:**
-  No modifications have been made to the model architecture for the training.
-  I fixed the orientation of the models in the dataset that were top view, then ran render.py to generate color and depth maps, then used openpose to get the keypoints and then used the op_json_to_numpy() function within the test_02_model.py script to convert the keypoints to numpy arrays to be saved in the keypoints folder under PERS .
-    I noticed that around 200 out of 14,000 samples in the train list were causing failures in the inv_affine function. Consequently, I decided to exclude these problematic samples from the training process. 

**Training Command:**
`NCCL_P2P_LEVEL=NVL OMP_NUM_THREADS=1 NCCL_SHM_DISABLE=1 CUDA_VISIBLE_DEVICES=0,1,2,3 python -m torch.distributed.run     --nnodes=1     --nproc_per_node=4     --rdzv_backend=c10d     train.py --use_ddp=True --data_path ./data --phase 1 --local_rank 0 --batch_size 1 --is_master True --num_epoch 30 --load_ckpt  '' --exp_name exp1 --workers 0`

**Loss on final epoch:** 1.2074

**Possible Reasons for the Issue?**
While I have identified the problematic samples and excluded them from training, I am uncertain if there might be other factors contributing to the unexpected results apart from the 200 lesser samples. Could you please provide insights on other potential reasons for the discrepancy between the expected and observed results during Phase 1 training? 

**Sub note:** Also training the phase 2 on the checkpoint from phase 1 results in a 'checkpoint.pth.tar' file that is only ~300mb compared to ~900mb pretrained checkpoint that comes with this repository and the ~900 mb checkpoint from phase 1. Can you also provide some insights on why this is happening?

Your guidance and expertise will be greatly appreciated in resolving this issue and achieving the desired training outcomes. Thank you for your assistance.

**Comment by SangHunHan92** [comment]
Thanks for sharing the results of your own experiments. The following is what I did to solve the problems I encountered while learning:

1. The first and most important thing is that the image must not have a background during normal prediction! Can you check if the background is removed from the image that enters the model during Training and Test?

2. Normal Map prediction results should generally be purple. But sometimes, like the first picture, there is a case of green. In this case, it was easy to judge after 500 to 1000 iterations, so in this case, learning was resumed from the beginning.

3. Removing about 200 samples did not have much effect. However, it was confirmed that performance is better when training multiple datasets such as RenderPeople or THuman2 together.

4. I know that checkpoint.pth.tar is reduced to 300mb after phase 2 because grad is removed from Normal Network. Since it is not used during inference, even 300mb ckpt works well.

Thank you.

**Comment by PremSuresh** [comment]
Thanks for the quick reply,
The images generated by render.py that I used for training have a black background as shown below. Should that be removed and replaced with a transparent background like it happens in test_02?
![00003_front](https://github.com/SangHunHan92/2K2K/assets/37246172/5f3285a5-4e58-4c8c-a00f-72c1463eb007)
![00003_front](https://github.com/SangHunHan92/2K2K/assets/37246172/180ae016-a0d8-4be5-92c8-6c169213984d)
Note: Thank you for the suggestions about checking the progress after 1000 iterations and restart training if i notice a green tint and using additional datasets. I'll be sure to try that as well.

**Comment by SangHunHan92** [comment]
Nope. I trained model with black background images. 
It's useful to check the training process through Tensorboard. 

Please check part-wise normal map output during training Phase 1. 
Background of normal map should also be black.

**Comment by GuiYu-wq** [comment]
@PremSuresh Hello, how long does it take you to render an image with cam_res=256 running? I am under cam_res=2048, and there is no response in 12 hours
